src/rgw/rgw_rest_s3.cc

```diff
@@ -2667,10 +2667,21 @@ int RGWCreateBucket_ObjStore_S3::get_params(optional_yield y)
 
 void RGWCreateBucket_ObjStore_S3::send_response()
 {
-  if (op_ret == -ERR_BUCKET_EXISTS)
-    op_ret = 0;
-  if (op_ret)
+  if (op_ret == -ERR_BUCKET_EXISTS) {
+    const auto eexist_override = s->cct->_conf.get_val<bool>("rgw_bucket_eexist_override");
+    if (! eexist_override) [[likely]] {
+      op_ret = 0;
+    } else {
+      /* n.b., EEXIST is mapped to HTTP error 409, BucketAlreadyExists, but ERR_BUCKET_EXISTS is not */
+      op_ret = -EEXIST;
```

Comment:
this is confusing, isn't it? maybe we should just add the same mapping for ERR_BUCKET_EXISTS?
```diff
     { EEXIST, {409, "BucketAlreadyExists" }},
+    { ERR_BUCKET_EXISTS, {409, "BucketAlreadyExists" }},
```

Reply:
will that not break anything?